src/App.tsx

```diff
@@ -1,73 +1,98 @@
+    setQuery(event.target.value);
+    setShowDropdown(false);
+    applyFilterQuery(event.target.value);
+  };
+
+  const handleClick = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setFilterQuery(person.name);
+    setShowDropdown(false);
+  };
+
+  const handleBlur = () => {
+    setTimeout(() => {
+      setShowDropdown(false);
+    }, 100);
+  };
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {selectedPerson?.name === filterQuery
+            ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+            : 'No selected person'}
         </h1>
 
-        <div className="dropdown is-active">
+        <div
+          className={`dropdown ${showDropdown && preparedPeople.length > 0 ? 'is-active' : ''}`}
+        >
           <div className="dropdown-trigger">
             <input
               type="text"
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
+              value={query}
+              onChange={handleQueryChange}
+              onFocus={() => setShowDropdown(true)}
+              onBlur={handleBlur}
             />
           </div>
 
           <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
             <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
-              </div>
+              {preparedPeople.map((person: Person) => (
+                <div
+                  className="dropdown-item"
+                  data-cy="suggestion-item"
+                  key={person.slug}
+                  onClick={() => handleClick(person)}
```

Comment:
```suggestion
                  onMouseDown={() => handleClick(person)}
```
and remove the setTImeout from the handleBlur, it should work